### PR TITLE
Fix #1380: Add join_paths and join prefix with include/lib dirs

### LINKED ISSUE #1380: [cmake] Incorrect CMake paths
I'm trying to package Raylib for NixOS and am running into some troubles with the pkgconfig.

I'm not well versed with CMake nor NixOS and so cannot provide much detail here. Suffice to say, I'm seeing a duplication of paths in the pkconfig output.

Here are two examples or the output:
From [contributor](https://github.com/NixOS/nixpkgs/pull/98030#pullrequestreview-488765959)
```
prefix=/nix/store/kxlh5s1p5ld9bxk22wrl1zpglx493apw-raylib-3.0.0
exec_prefix=${prefix}
libdir=${exec_prefix}//nix/store/kxlh5s1p5ld9bxk22wrl1zpglx493apw-raylib-3.0.0/lib
includedir=${prefix}//nix/store/kxlh5s1p5ld9bxk22wrl1zpglx493apw-raylib-3.0.0/include
```
My own workspace
```
prefix=/nix/store/zz0arpkbn47268lxllzq1rqsgp470d9b-raylib-3.0.0
exec_prefix=${prefix}
libdir=${exec_prefix}//home/adam/Workspace/nixpkgs-temp/raylib/out/lib
includedir=${prefix}//home/adam/Workspace/nixpkgs-temp/raylib/out/include
```

Effectively what I think I'm seeing is something like this:
```
prefix=/A
exec_prefix=/A
libdir=$prefix/A -> /A//A
includedir=$prefix/A -> /A//A
```

[Comments suggest](https://github.com/NixOS/nixpkgs/pull/98030#issuecomment-694765789) that this is actually an upstream issue and not actually Nix related (although it is being exposed by it).
It appears that the raylib.pc.in paths are incorrectly specified.

### PR DESCRIPTION
**Pull request by adamlwgriffiths**
Fixes incorrect usage of CMake paths which cause issues
on some more complicated environments (NixOS especially).

### CODE REVIEW

**Comment by adamlwgriffiths** [comment]
For some reason GH didn't link this to the issue, thought the title would have been enough.
I can re-push the commit with a reference to the issue if you'd like.

File: cmake/JoinPaths.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# This module provides function for joining paths
+# known from from most languages
```
Comment (a3f) [inline cmake/JoinPaths.cmake:2]: `s/from from/from/`

File: raylib.pc.in [hunk 2]
```diff
@@ -1,7 +1,7 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=${prefix}
```
Comment (a3f) [inline raylib.pc.in:2]: Why is changing this line necessary?

**Comment by a3f** [comment]
Apart from the comments, the changes look ok to me.

**Comment by raysan5** [comment]
@a3f thank you very much for reviewing, feel free to merge if you consider it ok.

File: cmake/JoinPaths.cmake (hunk 1 above)
Comment (adamlwgriffiths) [reply cmake/JoinPaths.cmake:2]: This was copy/pasted from https://github.com/jtojnar/cmake-snips/blob/master/CMakeScripts/JoinPaths.cmake
I've addressed the typo.

File: raylib.pc.in (hunk 2 above)
Comment (adamlwgriffiths) [reply raylib.pc.in:2]: This was copy pasted from https://github.com/fmtlib/fmt/blob/master/support/cmake/fmt.pc.in
I've reverted the change.

**Comment by adamlwgriffiths** [comment]
Addressed comments, should be g2g

**Comment by a3f** [comment]
Merged. Thanks, @adamlwgriffiths! Please let us know if you run into any further issues while packaging. :-)

**Comment by adamlwgriffiths** [comment]
Nps, glad to help

**Pull request merged**